test/test_unary_ufuncs.py

```diff
@@ -295,7 +308,39 @@ def test_reference_numerics(self, device, dtype, op):
                                            exact_dtype=exact_dtype, rtol=1e-3, atol=1e-2)
                     continue
 
-            self.assertEqualHelper(actual, expected, msg, dtype=dtype, exact_dtype=exact_dtype)
+            self.assertEqualHelper(actual, expected, msg, dtype=dtype, equal_nan=equal_nan, exact_dtype=exact_dtype)
+
+    # Tests that the function and its (array-accepting) reference produce the same
+    #   values on a range of tensors, including empty tensors, scalar tensors,
+    #   1D tensors and a large 2D tensor with interesting and extremal values
+    #   and discontiguities.
+    @suppress_warnings
+    @ops(unary_ufuncs)
+    def test_reference_numerics(self, device, dtype, op):
+        tensors = generate_numeric_tensors(device, dtype,
+                                           domain=op.domain)
+        self._test_reference_numerics(dtype, op, tensors)
+
+    @suppress_warnings
+    @ops(unary_ufuncs, allowed_dtypes=(*get_all_fp_dtypes(), *get_all_complex_dtypes(),
+                                       torch.int8, torch.int16, torch.int32, torch.int64))
+    def test_reference_numerics_hard(self, device, dtype, op):
+        if not op.handles_large_floats:
+            raise self.skipTest("This op does not handle large values")
+
+        tensors = generate_numeric_tensors_hard(device, dtype)
+        self._test_reference_numerics(dtype, op, tensors)
+
+    @suppress_warnings
+    @ops(unary_ufuncs, allowed_dtypes=(*get_all_fp_dtypes(), *get_all_complex_dtypes()))
+    def test_reference_numerics_extremal(self, device, dtype, op):
```

Comment:
The failures we're seeing make me think we should separate this test even further into "test_reference_numerics_extermal_{nonvectorized, vectorized}", because looking at the errors it seems like many operations are failing the vectorized version of this test, and it's unclear whether they would fail the nonvectorized version. If they do, however, it should provide a clearer error message since the helper function produces custom error messages for small tensors.

The test failures are believable to me. That is, it's credible that all those tests are failing and we should add skips as appropriate.